File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool containsOnlySpace(string str) {
    return all_of(str.begin(), str.end(), ::isspace);
}

int solution(vector<string> babbling) {
    int find = 0, npos = 0, answer = 0;
    string v[4] = {"aya", "ye", "woo", "ma"};
    for (int i = 0; i < babbling.size(); i++) {
        string temp = babbling[i];
        int j = 30, k = 0;
        while (j--) {
            npos = temp.find(v[k]);
            if (npos != string::npos) {
                temp.replace(npos, v[k].size(), " ");
                if (temp.find(v[k]) == npos + 1) {
                    break;
                }
            }
            k = (k + 1) % 4;
        }
        if (containsOnlySpace(temp)) answer++;
    }
    return answer;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
import java.util.*;

class Solution {
    
    static final String[] BABBLING = {"ye", "woo", "ma", "aya"};
    
    public int solution(String[] babbling) {
        int answer = 0;
        
        for (String str : babbling) {
            
            for (String bab : BABBLING) {
                int repeat = str.length() / bab.length();
                while (repeat >= 2) {
                    String temp = bab.repeat(repeat);
                    str = str.replaceAll(temp, "1");
                    repeat--;
                }
            }
            
            for (String bab : BABBLING) {
                str = str.replaceAll(bab, "2");
            }
            boolean check = true;
            for (int i = 0; i < str.length(); i++) {
                if (str.charAt(i) != '2') {
                    check = false;
                    break;
                }
            }
            if (check) answer++;
        }
        
        return answer;
    }
}